### DISCUSSION TITLE
Unable to connect to postgresql using Go and pq

### DISCUSSION THREAD

**Discussion by strong-728** [community/community#36990] [Other Feature Feedback, Questions, & Ideas] [answered]
I'm trying to connect a Go application with postgresql.

The app import postgresql driver:

"crypto/tls"
"database/sql"
"fmt"
"log"
"os"
"os/signal"
 ...
_ "github.com/go-sql-driver/mysql"
_ "github.com/lib/pq"
_ "github.com/mattn/go-sqlite3"
and uses like it to connect to the database:

driver, cnxn := dbFromURI(dbURI)
db, err := sql.Open(driver, cnxn)
if err != nil {
    panic(err)
}
and the dbFromUri method just split the info

func dbFromURI(uri string) (string, string) {
    parts := strings.Split(uri, "://")
    return parts[0], parts[1]
}
My URI works locally when i run the command : psql postgresql://user:user@172.20.0.1:5432/lcp

But in go i Go, I got

./lcpserver

I tried this URI for Go without success : psql postgresql://user:user@172.20.0.1:5432/lcp?sslmode=disable

Do you have any idea why i can't connect ? I tried with my aws rds postgres database, and got same result. Thnaks for the help.

complete code of the server https://github.com/readium/readium-lcp-server/blob/master/lcpserver/lcpserver.go

I change the typo and the demo provided i succeed the connexion. But in the lcp server i style got the same issue.

postgres://user:user@172.20.0.1:5432/lcp?sslmode=disable

**Comment by codeshare007** [comment, marked as answer]
Reference in https://stackoverflow.com/questions/66512318/unable-to-connect-to-postgresql-using-go-and-pq
